src/components/RecordMoviesComponent/index.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState } from "react";
+import classNames from "classnames";
+
+import RecordMoviesModal from "../RecordMoviesModal";
+import ViewportButton from "../ViewportButton";
+
+import styles from "./style.css";
+
+interface RecordMovieComponentProps {
+    movieUrl: string;
+    movieTitle: string;
+    cleanupMovieState: () => void;
+    startRecording: () => void;
+    stopRecording: () => void;
+}
+
+const RecordMovieComponent = (props: RecordMovieComponentProps) => {
+    const {
+        movieUrl,
+        movieTitle,
+        cleanupMovieState,
+        startRecording,
+        stopRecording,
+    } = props;
+    const [isRecording, setIsRecording] = useState<boolean>(false);
+    const [modalVisible, setModalVisible] = useState(false);
+    const [isHovering, setIsHovering] = useState(false);
+
+    const supportedBrowser = "VideoEncoder" in window;
+
+    const closeModal = () => {
+        cleanupMovieState();
+        setModalVisible(false);
+    };
+
+    const start = () => {
+        setIsRecording(true);
+        startRecording();
+    };
+
+    const stop = () => {
+        setIsRecording(false);
+        stopRecording();
+        setModalVisible(true);
+    };
+
+    const downloadMovie = () => {
+        if (!movieUrl) {
+            console.error("No recorded movie to download");
+            return;
+        }
+        const anchor = document.createElement("a");
+        anchor.href = movieUrl;
+        anchor.download = movieTitle;
+        document.body.appendChild(anchor);
+        anchor.click();
+        document.body.removeChild(anchor);
+        cleanupMovieState();
+    };
+
+    /**
+     * In this icon we are stacking glyphs to create multicolor icons via icomoon
+     */
+    const startRecordingIcon = (
+        <span className={styles.iconContainer}>
+            <span
+                className={classNames("icon-moon", "record-icon-circle")}
+            ></span>
+            <span
+                className={classNames("icon-moon", "record-icon-ring")}
+            ></span>
+        </span>
+    );
+
+    const activeRecordingIcon = (
+        <div className={classNames([styles.redCircle, styles.animate])}></div>
+    );
+
+    const stopRecordingIcon = isHovering
+        ? "stop-record-icon"
+        : activeRecordingIcon;
+
+    const getIcon = () => {
+        return isRecording ? stopRecordingIcon : startRecordingIcon;
+    };
```

Comment:
nit: but these two things seem really similar, but one is just a constant and one is a function? unless there is a reason I would pick one setup and go with it for both